x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_logic.ts

```diff
@@ -22,43 +22,77 @@ import {
   PostConnectorConfigurationArgs,
   PostConnectorConfigurationResponse,
 } from '../../../api/connector_package/update_connector_configuration_api_logic';
+import {
+  FetchIndexApiLogic,
+  FetchIndexApiParams,
+  FetchIndexApiResponse,
+} from '../../../api/index/fetch_index_api_logic';
+import { isConnectorIndex } from '../../../utils/indices';
+import { IndexViewLogic } from '../index_view_logic';
 
 type ConnectorConfigurationActions = Pick<
   Actions<PostConnectorConfigurationArgs, PostConnectorConfigurationResponse>,
   'apiError' | 'apiSuccess' | 'makeRequest'
 > & {
+  fetchIndexApiSuccess: Actions<FetchIndexApiParams, FetchIndexApiResponse>['apiSuccess'];
+  saveConfig: () => void;
   setConfigState(configState: ConnectorConfiguration): {
     configState: ConnectorConfiguration;
   };
   setIsEditing(isEditing: boolean): { isEditing: boolean };
+  setLocalConfigEntry(configEntry: ConfigEntry): ConfigEntry;
+  setLocalConfigState(configState: ConnectorConfiguration): {
+    configState: ConnectorConfiguration;
+  };
+  startFetchIndexPoll: () => void;
+  stopFetchIndexPoll: () => void;
 };
 
 interface ConnectorConfigurationValues {
   configState: ConnectorConfiguration;
+  configView: ConfigEntry[];
+  index: FetchIndexApiResponse;
   isEditing: boolean;
+  localConfigState: ConnectorConfiguration;
+  localConfigView: ConfigEntry[];
 }
 
-interface ConnectorConfigurationProps {
-  configuration: ConnectorConfiguration;
+interface ConfigEntry {
+  key: string;
+  label: string;
+  value: string;
 }
 
 export const ConnectorConfigurationLogic = kea<
-  MakeLogicType<
-    ConnectorConfigurationValues,
-    ConnectorConfigurationActions,
-    ConnectorConfigurationProps
-  >
+  MakeLogicType<ConnectorConfigurationValues, ConnectorConfigurationActions>
 >({
   actions: {
+    saveConfig: true,
     setConfigState: (configState: ConnectorConfiguration) => ({ configState }),
     setIsEditing: (isEditing: boolean) => ({
       isEditing,
     }),
+    setLocalConfigEntry: (configEntry: ConfigEntry) => ({ ...configEntry }),
+    setLocalConfigState: (configState: ConnectorConfiguration) => ({ configState }),
   },
   connect: {
-    actions: [ConnectorConfigurationApiLogic, ['apiError', 'apiSuccess', 'makeRequest']],
+    actions: [
+      ConnectorConfigurationApiLogic,
+      ['apiError', 'apiSuccess', 'makeRequest'],
+      FetchIndexApiLogic,
+      ['apiSuccess as fetchIndexApiSuccess'],
+      IndexViewLogic,
+      ['stopFetchIndexPoll', 'startFetchIndexPoll'],
```

Comment:
What are these being used for? Not clear to me.

Reply:
Nothing, I need to delete them, oops